### PR TITLE
Add support for MongoDB cluster using TLS

### LINKED ISSUE #6364: Support MongoDB TLS configuration
Hi,

I have two different mongo setups. One which is in a local kubernetes stack the other using cloud hosted Atlas.

Using presto i can connect fine to the Atlas version but not the local one. My local mongo is configured to only allow TLS connections and has its services available only via TLS. However, the TLS is signed by an internal CA.

I have imported the full mongo  ca chain to my keystore and then updated the following settings:

jmx.config
```
    -Djavax.net.ssl.trustStore=/etc/keystore/cacerts
    -Djavax.net.ssl.trustStoreType=pkcs12
    -Djavax.net.ssl.trustStorePassword=changeit
```

config.properties
```
    http-server.https.keystore.path=/etc/keystore/keystore.p12
    http-server.https.keystore.key=xxx
    http-server.https.truststore.path=/etc/keystore/cacerts
    http-server.https.truststore.key=xxx
    internal-communication.https.required=false
    internal-communication.https.keystore.path=/etc/keystore/keystore.p12
    internal-communication.https.keystore.key=xxx
    internal-communication.https.truststore.path=/etc/keystore/cacerts
    internal-communication.https.truststore.key=xxx
    workerInfo.http-client.trust-store-path=/etc/keystore/cacerts
    workerInfo.http-client.trust-store-password=xxx
    dynamic.http-client.trust-store-path=/etc/keystore/cacerts
    dynamic.http-client.trust-store-password=xxx
    discovery.http-client.trust-store-path=/etc/keystore/cacerts
    discovery.http-client.trust-store-password=xxx
```

  mongodb.properties: |
    connector.name=mongodb
    mongodb.seeds=mongodb-0.mongodb.databases.svc.cluster.local,mongodb-1.mongodb.databases.svc.cluster.local,mongodb-2.mongodb.databases.svc.cluster.local
    mongodb.credentials=user:password@admin
    mongodb.ssl.enabled=true
    mongodb.required-replica-set=rs0

I believe presto can find the servers ok, but I do not think it is presenting an TLS cert to mongo as I see a lot of the following errors in my mongo logs:
{"t":{"$date":"2020-12-16T16:26:55.304+00:00"},"s":"I",  "c":"NETWORK",  "id":22988,   "ctx":"conn15864","msg":"Error receiving request from client. Ending connection from remote","attr":{"error":{"code":141,"codeName":"SSLHandshakeFailed","errmsg":"no SSL certificate provided by peer; connection rejected"},"remote":"10.42.1.59:48560","connectionId":15864}}

In Presto logs i see no errors but if i try a call they just time out.

Any ideas on what i have missed welcome :)

### PR DESCRIPTION
**Pull request by ebyhr**
Fixes #6364

### CODE REVIEW

File: presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java [hunk 1]
```diff
@@ -50,6 +54,10 @@
     private int maxConnectionIdleTime;
     private boolean socketKeepAlive;
     private boolean tlsEnabled;
+    private File keystorePath;
```
Comment (kokosing) [inline presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java:57]: This should be captured as separate config class and it should be only binded when `tlsEnabled == true`

File: presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java [hunk 2]
```diff
@@ -50,6 +54,10 @@
     private int maxConnectionIdleTime;
     private boolean socketKeepAlive;
     private boolean tlsEnabled;
```
Comment (kokosing) [inline presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoClientConfig.java:56]: How it was working before?

**Review by kokosing — COMMENTED** [review]
Is it possible to add a real test with TLS secured mongodb instance?

**Comment by ebyhr** [comment]
Note: TLS options (https://docs.mongodb.com/manual/reference/connection-string/#tls-options) will be supported in #9819.

**Comment by ebyhr** [comment]
@akshayhshenoy The above options were already supported in the #9819.

**Comment by akshayhshenoy** [comment]
@ebyhr , I was going through the documentation and couldn't find a way to provide a tls certificate path to connect to mongoDB eg: "mongodb.tls.keystore-path" , Can you please point me to a location where I can get that information. Thanks!

**Comment by ebyhr** [comment]
@akshayhshenoy You can specify options in https://docs.mongodb.com/manual/reference/connection-string/#tls-options likes `mongodb.connection-url=...? tlsCertificateKeyFile =...`

**Comment by akshayhshenoy** [comment]
@ebyhr  Oh I see, Thanks for the information.

**Comment by udayabaski** [comment]
@ebyhr mongodb.connection-url=...? tlsCertificateKeyFile =... is not working. 

13:34:06.517 [main] WARN org.mongodb.driver.uri - Connection string contains unsupported option 'tlscafile'.
13:34:06.518 [main] WARN org.mongodb.driver.uri - Connection string contains unsupported option 'tlscertificatekeyfilepassword'.
13:34:06.518 [main] WARN org.mongodb.driver.uri - Connection string contains unsupported option 'tlscertificatekeyfile'.

Not sure why PR was closed. 

@akshayhshenoy Were you able to make it work?

**Pull request closed without merging**